src/docs/release/breaking-changes/mousetrackerannotation-requires-parameter-key.md

```diff
@@ -0,0 +1,177 @@
+---
+title: MouseTrackerAnnotation requires parameter "key"
+description: [Brief description similar to the "context" section below.]
+---
+
+# MouseTrackerAnnotation requires parameter "key"
+
+
+## Context
+
+Mouse events, such as when a mouse pointer has entered a region, exited, or
+is hovering a region, are detected with the help of
+`MouseTrackerAnnotation`s that are placed on interested regions during the
+render phase. Upon each new frame or new event, `MouseTracker` compares the
+annotations hovered by the mouse pointer before and after the change, then
+dispatches callbacks accordingly. `MouseTracker` also tracks which
+annotations are on the screen by requring the annotation's owners to attach
+or detach the annotation promptly.
+
+However, since the `MouseTrackerAnnotation` is a constant class, its properties
+can't be changed without replacing the annotation as a whole, which triggers
+the exit and enter callback if it's hovered by a mouse pointer.
+
+This issue is a blocker to the plan of supporting mouse cursor, since it’s
+very likely that the cursor assigned to a region needs to be changed.
+
+
+## Description of change
+
+Class `MouseTrackerAnnotation` now requires a new parameter `key`:
+
+```diff
+ class MouseTrackerAnnotation extends Diagnosticable {
+   const MouseTrackerAnnotation({
+     this.onEnter,
+     this.onHover,
+     this.onExit,
+-  });
++    @required this.key,
++  }) : assert(key != null);
+
++  final LocalKey key;
+
+   ...
+ }
+```
+
+Three of class `MouseTracker`'s methods, `attachAnnotation`,
+`detachAnnotation`, and `isAnnotationAttached`, instead of taking a
+single parameter of `MouseTrackerAnnotation`, will now take a
+single parameter `LocalKey`:
+
+```diff
+ class MouseTracker extends ChangeNotifier {
+   ...
+
+   @visibleForTesting
+-  bool isAnnotationAttached(MouseTrackerAnnotation annotation);
++  bool isAnnotationAttached(LocalKey annotationKey);
+
+-  void attachAnnotation(MouseTrackerAnnotation annotation);
++  void attachAnnotation(LocalKey annotationKey);
+
+-  void detachAnnotation(MouseTrackerAnnotation annotation);
++  void detachAnnotation(LocalKey annotationKey);
+ }
+```
+
+## Migration guide
+
+There are 2 ways of migrating affected code. The first way requires little
+work, while the second is much simpler when the annotation's properties
```

Comment:
"requires little work" means it's easy
"requires a little work" means it's not easy
I think this was supposed to be the second?

Reply:
Yes. I used "requires less work" and made the difference of the 2 methods clearer.